### PR TITLE
Speed control of the four motors

### LINKED ISSUE #12: Implement framework for controlling the ESCs
Implement framework for controlling the ESCs

### PR DESCRIPTION
**Pull request by jamie-54**
## Main Changes

* Updated `motors.cpp`
* Updated `motors.hpp`
* Updated `config.hpp`
* Updated `main.cpp` 
* Updated `.gitignore`

`motors.cpp` uses the RC_ESC library. First an object is created for each of our 4 ESC and assigned the correct pin, min and max PWM and arm speed from the ESC class.
Each object can be used to call the correct function from the ESC.h file under our functions. 
The writeSpeedToEsc function was updated to write the correct speed to the correct ESC.

`motors.hpp`. The attachMotors function was removed as the pins are attached to the correct motors in the `motors.cpp` file when the objects are created. The writeSpeedToEsc function was updated to take 4 inputs.

`config.hpp`. The correct ESC pins where updated for ESP32.

`main.cpp` has a test code for the motors, this should be moved to a new folder/file, maybe a `test_hardware` folder?

`.gitignore` added `.vscode/extensions.json`

## Issues

Closes #12

### CODE REVIEW

File: include/motors.hpp [hunk 1]
```diff
@@ -4,16 +4,11 @@
...
 namespace bzzz
-{
-
-    /**
-     *
-     */
-    void attachMotors(void);
+{   
     /**
      *
```
Comment (alphaville) [inline include/motors.hpp:9]: Documentation is missing

File: src/main.cpp [hunk 2]
```diff
@@ -1,29 +1,32 @@
 #include <Arduino.h>
-#include "controller.hpp"
+
```
Comment (alphaville) [inline src/main.cpp:2]: include config.hpp

File: src/main.cpp [hunk 3]
```diff
@@ -1,29 +1,32 @@
 #include <Arduino.h>
```
Comment (alphaville) [inline src/main.cpp:1]: Let's move this to `hardware/tests/hw_test_motors.cpp`

File: src/motors.cpp [hunk 4]
```diff
@@ -1,28 +1,40 @@
 
 #include "motors.hpp"
+#include "ESC.h"
```
Comment (alphaville) [inline src/motors.cpp:3]: what is `ESC.h`?

**Review by alphaville — CHANGES_REQUESTED** [review]

File: src/motors.cpp [hunk 5]
```diff
@@ -1,28 +1,40 @@
...
+    ESC backLeftEsc(BACK_LEFT_ESC_PIN, ABSOLUTE_MIN_PWM, ABSOLUTE_MAX_PWM, ARM_ROTOR_SPEED);
+    ESC backRightEsc(BACK_RIGHT_ESC_PIN, ABSOLUTE_MIN_PWM, ABSOLUTE_MAX_PWM, ARM_ROTOR_SPEED);
 
-    void attachMotors(void){
-        // todo
-    };
-
-    void writeSpeedToEsc(void){
-        // todo
+    void writeSpeedToEsc(int rotor_speed_front_left, int rotor_speed_front_right, int rotor_speed_back_left, int rotor_speed_back_right)
```
Comment (alphaville) [inline src/motors.cpp:14]: Do we need some sort of clipping here to prevent very high speeds?

Comment (alphaville) [reply src/motors.cpp:14]: and to prevent negative speeds too

File: include/motors.hpp (hunk 1 above)
Comment (jamie-54) [reply include/motors.hpp:9]: Yea I'll sort that out

File: src/main.cpp (hunk 3 above)
Comment (jamie-54) [reply src/main.cpp:1]: yea good idea

File: src/motors.cpp (hunk 4 above)
Comment (jamie-54) [reply src/motors.cpp:3]: Its the RC_ESC library. I should add a comment

File: src/motors.cpp (hunk 5 above)
Comment (jamie-54) [reply src/motors.cpp:14]: That is taken care of by the RC_ESC library. In the `motors.cpp` file when the object is created for each ESC, the correct pin, the maximum and minimum PWM and the arm speed are assigned. The `ESC.cpp` from the RC_ESC library then does the clipping based of the maximum and minimum PWM

File: src/main.cpp (hunk 2 above)
Comment (alphaville) [reply src/main.cpp:2]: If anything from `config.hpp` is needed in main, it should be included. Often the compiler figures out the missing dependencies, but it's good practice to include all necessary headers explicitly.

File: src/motors.cpp (hunk 4 above)
Comment (alphaville) [reply src/motors.cpp]: Ah, right!

File: src/main.cpp (hunk 2 above)
Comment (jamie-54) [reply src/main.cpp:2]: Ok, `main.cpp` doesn't directly use anything from `config.hpp`, that's why I didn't include it.
I'm going to move the `main.cpp` into the `hardware/tests/hw_test_motors.cpp` now

**Pull request merged**